code/modules/industrial_lift/tram/tram_remote.dm

```diff
@@ -0,0 +1,147 @@
+#define TRAMCTRL_INBOUND "inbound"
+#define TRAMCTRL_OUTBOUND "outbound"
+#define TRAMCTRL_FAST "fast"
+#define TRAMCTRL_SAFE "safe"
+
+/obj/item/tram_remote
+	icon_state = "tramremote_nis"
+	inhand_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	icon = 'icons/obj/device.dmi'
+	name = "tram remote"
+	desc = "A remote control that can be linked to a tram. This can only go well."
+	w_class = WEIGHT_CLASS_TINY
+	///desired tram direction
+	var/direction = TRAMCTRL_INBOUND
+	///fast and fun, or safe and boring
+	var/mode = TRAMCTRL_FAST
+	///weakref to the tram piece we control
+	var/datum/weakref/tram_ref
+	///cooldown for the remote
+	COOLDOWN_DECLARE(tram_remote)
+
+///set tram control direction
+/obj/item/tram_remote/attack_self_secondary(mob/user)
+	var/static/list/desc = list(TRAMCTRL_INBOUND = "< inbound", TRAMCTRL_OUTBOUND = "outbound >")
+	switch(direction)
+		if(TRAMCTRL_INBOUND)
+			direction = TRAMCTRL_OUTBOUND
+		if(TRAMCTRL_OUTBOUND)
+			direction = TRAMCTRL_INBOUND
+	update_appearance()
+	balloon_alert(user, "direction: [desc[direction]]")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+///set safety bypass
+/obj/item/tram_remote/CtrlClick(mob/user)
+	var/static/list/desc = list(TRAMCTRL_SAFE = "safe", TRAMCTRL_FAST = "fast")
+	switch(mode)
+		if(TRAMCTRL_SAFE)
+			mode = TRAMCTRL_FAST
+		if(TRAMCTRL_FAST)
+			mode = TRAMCTRL_SAFE
+	update_appearance()
+	balloon_alert(user, "mode: [desc[mode]]")
+
+/obj/item/tram_remote/examine(mob/user)
+	. = ..()
+	if(!tram_ref)
+		. += "There is an X showing on the display."
+		. += "Left-click a tram request button to link."
+		return
+	else
+		switch(direction)
+			if(TRAMCTRL_INBOUND)
+				. += "The arrow on the display is pointing inbound."
+			if(TRAMCTRL_OUTBOUND)
+				. += "The arrow on the display is pointing outbound."
+		switch(mode)
+			if(TRAMCTRL_FAST)
+				. += "The rapid mode light is on."
+			if(TRAMCTRL_SAFE)
+				. += "The rapid mode light is off."
+		if (!COOLDOWN_FINISHED(src, tram_remote))
+			. += "The number on the display shows [DisplayTimeText(COOLDOWN_TIMELEFT(src, tram_remote), 1)]."
+		else
+			. += "The display indicates ready."
+		. += "Left-click to dispatch tram."
+		. += "Right-click to toggle direction."
+		. += "CTRL-click to toggle safety bypass."
+
+/obj/item/tram_remote/update_overlays()
+	. = ..()
+	if(!tram_ref)
+		icon_state = "tramremote_nis"
+		return
+	switch(direction)
+		if(TRAMCTRL_INBOUND)
+			icon_state = "tramremote_ib"
+		if(TRAMCTRL_OUTBOUND)
+			icon_state = "tramremote_ob"
+	if(mode == TRAMCTRL_FAST)
+		. += mutable_appearance(icon, "tramremote_emag")
+
+/obj/item/tram_remote/attack_self(mob/user)
+	if (!COOLDOWN_FINISHED(src, tram_remote))
+		balloon_alert(user, "cooldown: [DisplayTimeText(COOLDOWN_TIMELEFT(src, tram_remote), 1)]")
```

Comment:
i forget, doesn't this fully display "minutes"/"seconds"? we really need a shortened version of this to just use m and s

Reply:
No, it displays the cooldown in deciseconds so I had to add DisplayTimeText